### PR TITLE
Create and use new template for step return value

### PR DESCRIPTION
**Pull request by juliasilge**
We've been chatting about a revamp of the recipes documentation in general, to use templating to reduce repetition and make it easier to change things in one place vs. _many_. This is a first step toward that, just making a template for what recipe steps return.

### CODE REVIEW

**Comment by juliasilge** [comment]
I also made a few tiny changes in some check return documentation, for a little more consistency.

File: R/downsample.R [hunk 1]
```diff
@@ -28,10 +28,7 @@
 #' @param target An integer that will be used to subsample. This
 #'  should not be set by the user and will be populated by `prep`.
 #' @param seed An integer that will be used as the seed when downsampling.
-#' @return An updated version of `recipe` with the new step
-#'  added to the sequence of existing steps (if any). For the
-#'  `tidy` method, a tibble with columns `terms` which is
-#'  the variable used to sample.
```
Comment (hfrick) [inline R/downsample.R:34]: For downsample and upsample the sentence for the `tidy()` method is now removed while in `newvalues.R` and `factor2string.R` the sentence was adapted and moved slightly. Is that on purpose?

File: R/newvalues.R [hunk 2]
```diff
@@ -26,14 +26,16 @@
...
-#'  `tidy` method, a tibble with columns `terms` (the
-#'  selectors or variables selected).
+#' @return An updated version of `recipe` with the new check added to the
+#'  sequence of any existing operations.
 #' @export
 #' @details This check will break the `bake` function if any of the checked
 #'  columns does contain values it did not contain when `prep` was called
 #'  on the recipe. If the check passes, nothing is changed to the data.
+#'
+#'  When you [`tidy()`] this step, a tibble with columns `terms` (the
```
Comment (hfrick) [inline R/newvalues.R:36]:
```suggestion
#'  When you [`tidy()`] this check, a tibble with columns `terms` (the
```

**Review by hfrick — APPROVED** [review]
Looks great! It will be great to get rid of the redundancy. I added a question about the documentation of the `tidy()` method but that's all :+1:

File: R/upsample.R [hunk 3]
```diff
@@ -28,10 +28,7 @@
 #' @param target An integer that will be used to subsample. This
 #'  should not be set by the user and will be populated by `prep`.
 #' @param seed An integer that will be used as the seed when upsampling.
-#' @return An updated version of `recipe` with the new step
-#'  added to the sequence of existing steps (if any). For the
-#'  `tidy` method, a tibble with columns `terms` which is
-#'  the variable used to sample.
```
Comment (hfrick) [inline R/upsample.R:34]: this is the other place where the sentence on the `tidy()` method is removed

File: R/downsample.R (hunk 1 above)
Comment (juliasilge) [reply R/downsample.R:34]: Yes, these have been moved to themis and are deprecated here in recipes, soon to go away here entirely.

File: R/upsample.R (hunk 3 above)
Comment (juliasilge) [reply R/upsample.R:34]: Yes, since the "real" documentation now lives in themis 👍

**Comment by juliasilge** [comment]
Thank you so much @hfrick! 💟

**Pull request merged**